src/main/java/edu/tamu/sage/service/SimpleProcessorService.java

```diff
@@ -84,17 +84,11 @@
 
                     solrReader.getFields().forEach(field -> {
                         if (doc.getFieldValues(field.getName()) != null) {
-                            boolean hasValue = false;
-                            doc.getFieldValues(field.getName()).forEach(result -> {
-                                if (hasValue) {
-                                    return;
-                                }
-                                if (field.getSchemaMapping().equals("terms.identifier") && doc.getFieldValue(field.getName()).toString().contains("://")) {
-                                    resultsMap.put(field.getSchemaMapping(), new String(Base64.encodeBase64(result.toString().getBytes())));
-                                } else {
-                                    resultsMap.put(field.getSchemaMapping(), result);
-                                }
-                            });
+                            if (doc.getFieldValues(field.getName()).size() > 1) {
```

Comment:
This just makes more sense. I do feel the logic can be improved.

```
if (doc.containsKey(field.getName()) {
  Collection<Object> fields = doc.getFieldValues(field.getName());
  // would be nice to condition whether the destination Solr field is multi-value or not
  // we are assuming destination field will be multi-value if the source field had multiple values
  if (fields.size() > 1) {
    resultsMap.put(field.getSchemaMapping(), fields);
  } else if (field.size() == 1) {
    resultsMap.put(field.getSchemaMapping(), fields.get(0));
  }
}
```

Reply:
For future planning, I suggest we perform no logic here and just map the fields whatever they might be. Leave it to the writer to determine how the fields should be written.